### PR TITLE
[Heap] Improve type-level doc comment

### LINKED ISSUE #324: `Heap` documentation does not explain what a heap is
The current type documentation for `Heap` links to wikipedia for a definition of a min-max heap. This isn't ideal; rather, the doc comment needs to directly include a succinct description of what a heap is (in terms that are familiar to Swift engineers), including some rudimentary code samples exercising its operations.

rdar://118368503

### PR DESCRIPTION
**Pull request by lorentey**
Resolves #324.

rdar://118368503

### Checklist
- [X] I've read the [Contribution Guidelines](/README.md#contributing-to-swift-collections)
- [X] My contributions are licensed under the [Swift license](/LICENSE.txt).
- [X] I've followed the coding style of the rest of the project.
- [ ] I've added tests covering all new code paths my change adds to the project (if appropriate).
- [ ] I've added benchmarks covering new functionality (if appropriate).
- [X] I've verified that my change does not break any existing tests or introduce unexplained benchmark regressions.
- [X] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci test

**Comment by lorentey** [comment]
macOS PR tests are expected to fail due to rdar://117169742. This is fine. 🔥

**Comment by lorentey** [comment]
@jmschonfeld @glessard @Azoy This is a quick attempt at writing some minimal docs -- fresh eyes are welcome!

File: Sources/HeapModule/Heap.swift [hunk 1]
```diff
@@ -9,20 +9,54 @@
...
+///
+/// Unlike most container types, `Heap` doesn't provide a direct way to iterate
+/// over the elements it contains -- it isn't a `Sequence` (nor a `Collection`).
+/// This is because the order of items in a heap is unspecified and unstable:
+/// it may vary between heaps that contain the same set of items, and it may
+/// sometimes change in between versions of this library. In particular, the
+/// items are (almost) never expected to be in sorted order.
+///
+/// For cases where you do need to access the contents of a heap directly and
+/// you don't care about their (lack of) order, you can still do so invoking the
```
Comment (stephentyrone) [inline Sources/HeapModule/Heap.swift:50]:
```suggestion
/// you don't care about their (lack of) order, you can do so by invoking the
```

**Review by stephentyrone — APPROVED** [review]

**Review by jmschonfeld — APPROVED** [review]
LGTM!

**Comment by lorentey** [comment]
@swift-ci test

**Pull request merged**